📝(oauth2) Add documentation for moodle oauth2 login (#334)
openfun/openedx-docker: docs/moodle-oauth-configuration.md

# Moodle OAuth Configuration

> To integrate Moodle and edX, you can use OAuth to allow users to log in to Moodle using their Open EDX credentials.

Reviewer: ```suggestion
To integrate Moodle and Open edX, you can use OAuth to allow users to log in to Moodle using their Open EDX credentials.
```

> To integrate Moodle and OpenEdX, you can use OAuth to allow users to log in to Moodle using their OpenEdX credentials.

Author: :sweat_smile: please pick one @lunika @sampaccoud

> To integrate Moodle and OpenEdX, you can use OAuth to allow users to log in to Moodle using their OpenEdX credentials.

Reviewer: On the website https://openedx.org/ they are using `Open edX`. But maybe if it's the software we are talking about we should use `OpenEdX` @sampaccoud  ?

> To integrate Moodle and OpenEdX, you can use OAuth to allow users to log in to Moodle using their OpenEdX credentials.

Author: My 2 cents: 
```
nc@xps15 ~/p/f/openedx-docker (moodle-oauth2-documentation)> ag 'Open edX' -w | wc -l
35
nc@xps15 ~/p/f/openedx-docker (moodle-oauth2-documentation)> ag OpenEdX -w | wc -l
18
```

> To integrate Moodle and OpenEdX, you can use OAuth to allow users to log in to Moodle using their OpenEdX credentials.

Author: We have a clear winner on edx-platform repo:
```
nc@xps15 ~/p/f/edx-platform (master)> ag 'Open edX' -w | wc -l
1218
nc@xps15 ~/p/f/edx-platform (master)> ag OpenEdX -w | wc -l
48
```

> To integrate Moodle and OpenEdX, you can use OAuth to allow users to log in to Moodle using their OpenEdX credentials.

Author: I used `Open edX`

> Follow these steps to configure Moodle OAuth with edX.

Reviewer: ```suggestion
Follow these steps to configure Moodle OAuth with Open edX.
```

> ## Open edX Configuration
> 
> 1. In the Open edX platform, go to the Django admin panel at `http://<your LMS URL>/admin`.

Reviewer 2: We usually spell in OpenEdX
```suggestion
## OpenEdX Configuration

1. In the OpenEdX platform, go to the Django admin panel at `http://<your LMS URL>/admin`.
```

2. Go to http://localhost:8072/admin/oauth2/client/ and click **Add Client**.
3. Fill the **URI** field with the URL of your Moodle instance.

   `https://sandbox.moodledemo.net`
4. Fill the **Redirect uri** field with the callback URL of your Moodle instance.

   `https://sandbox.moodledemo.net/admin/oauth2callback.php`
5. Keep the generated client id and secret for later use.
6. Set the **client type** to `Confidential`.
7. Click **Save**.

![images/edx_oauth2_client.png](images/edx_oauth2_client.png)

## Moodle Configuration

[Detailed Moodle configuration documentation](https://docs.moodle.org/402/en/OAuth_2_services)

> ### Create a new oAuth2 custom service

Reviewer: ```suggestion
### Create a new OAuth2 custom service
```

1. In the Moodle admin panel, go to `Site administration > Server > OAuth 2 services`.
2. Click **Create a new custom service**.
3. Fill the **Name** field with a name of your choice.

> 4. FIll the **Client ID** and **Client secret** fields with the values generated in the Open edX configuration step.

Reviewer: ```suggestion
4. Fill the **Client ID** and **Client secret** fields with the values generated in the Open edX configuration step.
```

> 5. Fill the **Service base URL** field with the URL of your Open edX instance.

Reviewer 2: ```suggestion
5. Fill the **Service base URL** field with the URL of your OpenEdX instance.
```

>    `http://<your LMS URL>`

Reviewer 2: Since both are LMSes;
```suggestion
   `http://<your openEdX LMS URL>`
```

6. Set the **This service will be used** field to `Login page only`.
7. Fill the **Name displayed on the login page** field with a name of your choice.
8. Uncheck the **Require email verification** field.
9. Check the **I understand that disabling email verification can be a security issue.** field.
10. Click **Save changes**.

![Moodle admin: OAuth2 ](images/moodle_oauth2_service.png)

### Configure endpoints

1. Click on the **Configure endpoints** button.

   ![Moodle admin: navigate to webservice configure endpoints](images/moodle_oauth2_configure_endpoints.png)
2. Click on **Create new endpoint for issuer "Open EDX"**.
3. Create the following endpoints:

   | Name                   | URL                                          |
   |------------------------|----------------------------------------------|

>    | authorization_endpoint | `http://<your LMS URL>/oauth2/authorize/`    |
>    | token_endpoint         | `http://<your LMS URL>/oauth2/access_token/` |
>    | userinfo_endpoint      | `http://<your LMS URL>/oauth2/user_info/`    |

Reviewer 2: ```suggestion
   | authorization_endpoint | `http://<your OpenEdX LMS URL>/oauth2/authorize/`    |
   | token_endpoint         | `http://<your OpenEdX LMS URL>/oauth2/access_token/` |
   | userinfo_endpoint      | `http://<your OpenEdX LMS URL>/oauth2/user_info/`    |
```

### Configure user field mapping

1. Click on the **Configure user field mappings** button.

   ![moodle_oauth2_configure_mappings.png](images/moodle_oauth2_configure_mappings.png)
2. Create the following mappings:

   | External field name | Internal field name |
   |---------------------|---------------------|
   | email               | email               |
   | preferred_username  | username            |

> ### Activate Oauth2 login

Reviewer: ```suggestion
### Activate OAuth2 login
```
